src/common/tiglcommonfunctions.cpp

```diff
@@ -1547,14 +1549,23 @@ bool IsPointInsideShape(const TopoDS_Shape &solid, gp_Pnt point)
         throw tigl::CTiglError("The shape is not a solid");
     }
 
+    // first, check if the point is in the bounding box
+    Bnd_Box boundingBox;
```

Comment:
How much time takes the creation of the bounding box itself? If this takes a major time, we could also cache the bounding box, right?

Reply:
Good idea! It is used in several places in TiGL, so this definitely makes sense. But it is probably not very expensive...

Reply:
Ok, so I had a first go at a cached bounding box in bd6a4cf327deb8c8de07cf146278ee3a9197ed48. Is this what you had in mind? My implementation seems to have a negligible effect:

```
[ RUN      ] TestPerformance.tiglCheckPointInside
Time tiglCheckPointInside wing [us]: 2620
Time tiglCheckPointInside vtp [us]: 1700
Time tiglCheckPointInside fuselage [us]: 337160
```

So it's up to you, @rainman110: Cached bounding box or not? If we go for the cached bounding box, we should remember to use it at all other places in tigl, where currently the bounding box is recalculated.

Reply:
In the tests, could you add a cold-start, that creates all shapes and bounding boxes? the time measured should not include these one-time effects.

Then it might be easier to see the impact of the performance optimization.

Reply:
Ok, more measurements with precalculated lofts and bounding boxes:

### No Bounding Box Check

```
[ RUN      ] TestPerformance.tiglCheckPointInside
Time tiglCheckPointInside wing [us]: 8220
Time tiglCheckPointInside vtp [us]: 6560
Time tiglCheckPointInside fuselage [us]: 1.16324e+06
```

### Bounding Box Check without bounding box cache:

```
[ RUN      ] TestPerformance.tiglCheckPointInside
Time tiglCheckPointInside wing [us]: 2720
Time tiglCheckPointInside vtp [us]: 1800
Time tiglCheckPointInside fuselage [us]: 320400
```

### Bounding Box Check with bounding box cache:

```
[ RUN      ] TestPerformance.tiglCheckPointInside
Time tiglCheckPointInside wing [us]: 2280
Time tiglCheckPointInside vtp [us]: 1540
Time tiglCheckPointInside fuselage [us]: 294540
```

Reply:
@rainman110, I just merged the cache into this PR, the test results speak for themselves